yt/visualization/base_plot_types.py

```diff
@@ -341,7 +341,19 @@ def _set_axes(self) -> None:
         self.image.axes.set_facecolor(self.colorbar_handler.background_color)
 
         self.cax.tick_params(which="both", direction="in")
-        self.cb = self.figure.colorbar(self.image, self.cax)
+
+        # For creating a multipanel plot by ImageGrid
+        # we may need the location keyword, which was introduced since Matplotlib 3.7.0
+        cb_location = getattr(self.cax, "orientation", None)
```

Comment:
I'm not sure I get why the "orientation" attribute is directly used as a "location" value. What are the possible values for `self.cax.orientation` ?

Reply:
In MPL, `ImageGrid.cbar_axes.orientation` is somehow designed to store the value from `ImageGrid.cbar_location` (see the [code](https://github.com/matplotlib/matplotlib/blob/a254b687df97cda8c6affa37a1dfcf213f8e6c3a/lib/mpl_toolkits/axes_grid1/axes_grid.py#L414C17-L414C28)), so that the value of `self.cax.orientation` here is supposed to be `top`,`right` , etc.

Reply:
I see. This is surprising but kinda out of our reach. Thanks !